redis/src/cluster_async/mod.rs

```diff
@@ -2342,6 +2503,7 @@ pub trait Connect: Sized {
         connection_timeout: Duration,
         socket_addr: Option<SocketAddr>,
         push_sender: Option<mpsc::UnboundedSender<PushInfo>>,
+        disconnect_notifier: Option<Box<dyn DisconnectNotifier>>,
```

Comment:
Why the `Box` + `dyn` is this because we need to support non `tokio` connections? (there is a performance penalty for dynamic dispatch that's why I am asking)

Reply:
Yep, this is correct

Reply:
We should take our time to do a proper cleanup of this library, since it contains many un-needed code sections that are increasing our work effort

Reply:
Definitely we should take an action to wipe out all the deprecated/irrelevant portions. It will make the development much easier and sustainable